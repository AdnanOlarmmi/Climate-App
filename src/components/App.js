import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter as Router,
  Routes, Route,

} from 'react-router-dom';
import { countriesInfo } from '../redux/country/countrySlice';
import HomePage from './countriesComponents/HomePage';
import DetailsPage from './countriesComponents/DetailsPage';
import Header from './countriesComponents/Header';

const App = () => {
  const countriesdata = useSelector((state) => state.countrydata);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(countriesInfo());
  });

  return (
    <Router>
      <Header />
      <Routes>
        <Route path="/" element={<HomePage countriesdata={countriesdata} />} />
        {countriesdata.map((countrydata) => (
          <Route
            key={countrydata.id}
            path={countrydata.ctryinfo.ctryname}
            element={<DetailsPage details={countrydata} />}
          />
        ))}
      </Routes>
    </Router>
  );
};

export default App;
