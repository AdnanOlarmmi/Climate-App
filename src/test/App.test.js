import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../components/App';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('My HomePage', () => {
  it('render My HomePage Component', () => {
    render(<App />);
    expect(App).toMatchSnapshot();
  });
});

export default render;