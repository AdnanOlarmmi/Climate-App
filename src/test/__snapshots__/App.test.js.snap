// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My HomePage render My HomePage Component 1`] = `[Function]`;
